homework2/hw5.py

```diff
@@ -0,0 +1,41 @@
+"""
+Some of the functions have a bit cumbersome behavior when we deal with
+positional and keyword arguments.
+
+Write a function that accept any iterable of unique values and then
+it behaves as range function:
+
+
+import string
+
+
+assert = custom_range(
+    string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(
+    string.ascii_lowercase, 'g', 'p'
+    ) == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(
+    string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+
+"""
+from typing import Any, List
+
+
+def custom_range(input_sequence: List[Any], *args) -> List[Any]:
+    """
+    Implementing a custom range function
+    which allows to iterate through any iterable object
+    with unique values.
+    """
+    range_args = []
+    if len(args) > 3:
```

Comment:
To simplify idea you can create
`def generate_cusmtom_range(input_sequence, start=None, stop=None, step=None)`